### PR TITLE
[newrelic-infrastructure] Add Windows support

### LINKED ISSUE #41: newrelic-infrastructure Windows support
**Is your feature request related to a problem? Please describe.**

Currently the `newrelic-infrastructure` Helm chart can only be run on Linux nodes. With the introduction of Windows nodes in EKS & GKE, it would be nice to have a version of this chart that can be run on Windows nodes. I see a version of the Docker `newrelic/infrastructure-k8s` image is now available with [windows-1909](https://hub.docker.com/layers/newrelic/infrastructure-k8s/1.21.0-windows-1909-alpha/images/sha256-3aee9ca7b4a7a28af87b9ee32f428933c1f9db875b72543b0bc8b157b3b383f6?context=explore) and [windows-1809](https://hub.docker.com/layers/newrelic/infrastructure-k8s/1.21.0-windows-1809-alpha/images/sha256-b56527942ac1bc1e5b4f517d92f6dcc613922502f05607e43c23a850a0f2d16e?context=explore) variants (in alpha).

**Describe the solution you'd like**

A version of the `newrelic-infrastructure` Helm chart that can be run on Windows nodes.

**Describe alternatives you've considered**

I've gone down the path of patching the existing chart with Windows alternatives - ripping out things like `hostNetwork`, which is currently not supported in Windows. In my mind, there should really be an official New Relic version though.

**Additional context**

New Relic Kubernetes integration is a `helm install` command away for Linux nodes, but not quite as straight and forward for Windows.

### PR DESCRIPTION
**Pull request by mr-smithers-excellent**
#### Is this a new chart?
No

#### What this PR does / why we need it:
Adds Windows support for New Relic Infrastructure on Windows nodes and sets sensible default values

#### Which issue this PR fixes:
  - Fixes #41 

#### Special notes for your reviewer:
Hi @alejandrodnm, here's what I've included in the PR:
* New Windows `DaemonSet` added (`daemonset-windows.yaml`)
* New values added and documented: `enableLinux`, `enableWindows`, `windowsNodeSelector` and `image.windowsTag`
* Added `.gitignore` from official Helm Charts repo
* The new `nrStaging` value was throwing an error from `_helpers.tpl` when global variables were not applied. I added some additional checking to fix this error.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

**Comment by alejandrodnm** [comment]
Hi @mr-smithers-excellent thanks for the PR, I'll try to review this today, or tomorrow. 

Also, thanks for spotting the `nrStaging` bug, in all my tests I had global sets and didn't spot it. At first I was trying to do something like what you did but in some cases it wasn't working so I did the check with `eq`. In your code, if you set any of the variables to `false` the template returns `'     false'` and it takes that as `true` in the `if`. I added a fix, we just needed to add the `-` to trim the values, it's already on master so you have to rebase.

**Comment by mr-smithers-excellent** [comment]
@alejandrodnm sounds good, I'll resolve that conflict. Looks like there's a conflict with the Chart version as well. What version would you like me to set here?

**Comment by alejandrodnm** [comment]
I didn't have time to review it today, I'll do it tomorrow.

As for the version you can increase to `0.15.0`, I'll talk with the team to latter move this to `1.0.0`

**Comment by mr-smithers-excellent** [comment]
No worries. Handled the rebasing and Chart version bump. All ready for review now.

File: charts/newrelic-infrastructure/templates/_helpers.tpl [hunk 1]
```diff
@@ -35,6 +35,7 @@ chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
 heritage: {{.Release.Service }}
 release: {{.Release.Name }}
 mode: {{ template "newrelic.mode" . }}
+{{- if .Values.windowsEnabled -}}kubernetes.io/os: windows{{- end }}
```
[the remark this answers is not in the document]
Comment (mr-smithers-excellent) [reply charts/newrelic-infrastructure/templates/_helpers.tpl:38]: Good point - I'll remove the label from this template. As you said, should be sufficient that it's already in the Windows DaemonSet.

**Comment by mr-smithers-excellent** [comment]
@alejandrodnm changes have been applied and CI is passing after another chart version bump.

**Comment by alejandrodnm** [comment]
@mr-smithers-excellent Thanks again for the PR :)

**Pull request merged**